tests/test_affine.py

```diff
@@ -60,6 +60,13 @@
                 p(np.array([[[0.0, 0.0, 0.0, 0.0], [0.0, 2.0, 0.0, 0.0], [0.0, 3.0, 1.0, 0.0], [0.0, 0.0, 0.0, 0.0]]])),
             ]
         )
+        TESTS.append(
+            [
+                dict(rotate_params=[np.pi / 2], padding_mode="zeros", device=device, align_corners=False),
+                {"img": p(np.arange(4).reshape((1, 2, 2))), "spatial_size": (4, 4)},
+                p(np.array([[[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 2.0, 0.0], [0.0, 0.0, 3.0, 1.0], [0.0, 0.0, 0.0, 0.0]]])),
+            ]
+        )
         TESTS.append(
```

Comment:
Hi @wyli , I found that when `align_corners=False`. lazy output always cannot get expected results. I took sometime but cannot find the root cause so far. Could you please help to provide some suggestions? Thanks!

Reply:
`peek_pending_affine` and `peek_pending_shape` are correct for pending output, but after calling `apply_transforms`, the results are different.

Reply:
the same issue happens in `zoom` when the mode is "nearest" which can create the correct affine and shape but the results after `apply_transforms` are different.

Reply:
Hi @yiheng-wang-nv, I've fixed the `align_corners=False` case, yesterday's commit was wrong, so I force pushed https://github.com/Project-MONAI/MONAI/pull/5860/commits/a43fe9f1c08ade5fa0a201bcca84076340af294b

@KumoLiu could you please let me know which lines cause the issue? I can take another look to double-check.

Reply:
Hi @wyli, I just tried to change the mode to `nearest` in L41 and L51, and I find that it could generate the correct shape and affine. But after `apply_transforms`, it can't give the correct output with non-lazy mode. I also output the results to see and discovered that the results will always have a few pixels that are not the same, most at the edge.
https://github.com/Project-MONAI/MONAI/blob/a43fe9f1c08ade5fa0a201bcca84076340af294b/tests/test_zoom.py#L41

Reply:
I've finished revising the relevant align corners options, and added consistency tests of Affine vs Rot90 and Resize:

https://github.com/Project-MONAI/MONAI/blob/916f7ced5151d8fb7486aa17667b6162c7889b83/tests/test_rotate90.py#L123-L125

https://github.com/Project-MONAI/MONAI/blob/916f7ced5151d8fb7486aa17667b6162c7889b83/tests/test_affine.py#L188-L191